examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,407 @@
+import typer
+import json
+import logging
+import langroid as lr
+from rich.prompt import Prompt, Confirm
+from typing import List, Tuple, Optional, Literal, Any
+from langroid.language_models import OpenAIGPTConfig
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.agent.task import Task
+from langroid.agent.tools.google_search_tool import GoogleSearchTool
+from langroid.agent.tools.metaphor_search_tool import MetaphorSearchTool
+from langroid.utils.logging import setup_logger
+from langroid.utils.configuration import set_global
+from config import get_base_llm_config, get_global_settings
+from models import SystemMessages, Message
+
+DEFAULT_SYSTEM_MESSAGE_ADDITION = f"""
+            PLEASE PROVIDE REFERENCES WITH URLS FOR ALL YOUR ARGUMENTS. 
```

Comment:
We shouldn't ask the LLM to produce URLs, they will almost never be reliable, unless they are well-known top-level names. Also, I don't think you're asking the feedback agent to verify the URLs, so best to drop.
It's best to not even say "ensure real sources" when we are not giving it access to docs. The best it can do is to produce text from its own baked-in "knowledge" in its weights.

Reply:
Cleaned up the prompt.